crates/trie/sparse/src/traits.rs

```diff
@@ -83,7 +73,25 @@ pub trait SparseTrieInterface: Sized + Debug + Send + Sync {
         path: Nibbles,
         node: TrieNode,
         masks: TrieMasks,
-    ) -> SparseTrieResult<()>;
+    ) -> SparseTrieResult<()> {
+        self.reveal_nodes(vec![RevealedSparseNode { path, node, masks }])
+    }
+
+    /// Reveals one or more trie nodes if they have not been revealed before.
+    ///
+    /// This function decodes trie nodes and inserts them into the trie structure. It handles
+    /// different node types (leaf, extension, branch) by appropriately adding them to the trie and
+    /// recursively revealing their children.
+    ///
+    /// # Arguments
+    ///
+    /// * `nodes` - The nodes to be revealed, each having a path and optional set of branch node
+    ///   masks. The nodes will be unsorted.
+    ///
+    /// # Returns
+    ///
+    /// `Ok(())` if successful, or an error if any of the nodes was not revealed.
+    fn reveal_nodes(&mut self, nodes: Vec<RevealedSparseNode>) -> SparseTrieResult<()>;
```

Comment:
we can benefit from accepting an iterator here, so that we don't do a vec heap allocation in `reveal_node` and instead just use `core::iter::once`. Can be done later in a separate PR.